DesiMock

```diff
@@ -0,0 +1,195 @@
+        nearest_peak_ix = np.argmin(peak_difference_spectrum)
+
+        # get the column density of the identfied nearest peak
+        _, potential_lya_nhi, _, _ = \
+            self.prediction.get_coldensity_for_peak(self.prediction.peaks_ixs[nearest_peak_ix])
+        _, potential_lyb_nhi, _, _ = \
+            self.prediction.get_coldensity_for_peak(peakix)
+
+        # Validations: check that the nearest peak is close enough to match
+        #              sanity check that the LyB is at least 0.3 less than the DLA
+        is_nearest_peak_within_range = peak_difference_spectrum[nearest_peak_ix] <= 15
+        is_nearest_peak_larger_coldensity = potential_lyb_nhi < potential_lya_nhi - 0.3
+
+        return is_nearest_peak_within_range and is_nearest_peak_larger_coldensity
+
+
+    def get_lyb_index(self, peakix):
+        """
+        Returns the index location of the Ly-B absorption for a given peak index value
+        :param peakix:
+        :return: index location of Ly-B
+        """
+        spectrum_higher = 10**self.loglam[peakix]
+        spectrum_lambda_lower = spectrum_higher * (1025.722 / 1215.67)
+        log_lambda_lower = np.log10(spectrum_lambda_lower)
+        ix_lambda_lower = (np.abs(self.loglam - log_lambda_lower)).argmin()
+        return ix_lambda_lower
+
+    def process(self, model_path):
+        """  The following should follow the algorithm in process_catalog
+        :param model_path:
+        :return:
+        """
+        import json
+        from dla_cnn.data_loader import scan_flux_sample
+        from dla_cnn.localize_model import predictions_ann as predictions_ann_c2
+        from dla_cnn.data_loader import compute_peaks, get_lam_data
+        #from dla_cnn.data_loader import add_abs_to_sightline
+        from dla_cnn.absorption import add_abs_to_sightline
+        from dla_cnn.data_model.Prediction import Prediction
+        # Fluxes
+        fluxes = scan_flux_sample(self.flux, self.loglam, self.z_qso, -1, stride=1)[0]
+        # Model
+        with open(model_path+"_hyperparams.json",'r') as f:
+            hyperparameters = json.load(f)
+        loc_pred, loc_conf, offsets, density_data_flat = predictions_ann_c2(hyperparameters, fluxes, model_path)
+        self.prediction = Prediction(loc_pred=loc_pred, loc_conf=loc_conf, offsets=offsets, density_data=density_data_flat)
+        # Peaks
+        _ = compute_peaks(self)
+        # Absorbers?
+        add_abs_to_sightline(self)
+
+
+class DesiMock:
+    """
+    a class to load all spectrums from a Desi fits file.
+    :attribute wavelength array-like, save the wavelength of all spectrum (all spectrum have same wavelength array)
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux, 'ERROR':error, 'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas'information}
```

Comment:
Describe the format of `dlas information`.

Reply:
`dla information` is a tuple, and its format is like (dla_id, z_qso, NHI).